### ISSUE TITLE
[bug] Execution not caching

### ISSUE THREAD

**Issue by Itega** [kubeflow/pipelines#5762]
### What steps did you take:
- [Deploy Kubeflow Pipelines on a new GKE cluster](https://www.kubeflow.org/docs/components/pipelines/installation/standalone-deployment/#deploying-kubeflow-pipelines)
- Run the [Kubeflow caching sample](https://github.com/kubeflow/pipelines/blob/master/samples/core/caching/caching_sample.py)

### What happened:
The sample failed with error :  
`RuntimeError: The cached execution was not re-used or pipeline run took to long to complete.`

### What did you expect to happen:
The sample should have run successfully. 

### Environment:
Kubernetes : 1.18.18-gke.1100
KFP version : 1.6.0
KFP SDK version: 1.6.2

**Comment by Bobgy** [comment]
@Ark-kun is this sample still working properly?
It fails in our test infra too:
https://testgrid.k8s.io/googleoss-kubeflow-pipelines#kubeflow-pipeline-postsubmit-integration-test

**Comment by Itega** [comment]
I can give you more information about my case as I have now found a way to install KFP 1.6 with caching on GCP. 
My problem was that I was using a GKE private cluster, I tried with a public cluster and it's working fine. 
Did some search and Kubeflow does not seems to be compatible with private cluster on GKE : https://github.com/kubeflow/gcp-blueprints/issues/267 Maybe you have more informations on this ?

Can confirm the sample is working well on my public cluster so I really think there is a problem with caching on some Kubernetes cluster configurations.
